bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -596,6 +596,11 @@ public static Object getAnnotValue(TypedescValue typedescValue, BString annotTag
         if (!(describingType instanceof BAnnotatableType)) {
             return null;
         }
+        if (describingType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG &&
+                ((BTypeReferenceType) describingType).getReferredType().getTag() == TypeTags.TUPLE_TAG) {
```

Comment:
What if the referred type is again typeRef?

Reply:
I think we can remove this check now, will test and update.

Reply:
Sorry I was mistaken we still need this at runtime type checker. 
However if the referred type is a type reference `getAnnotations` will give the annotations for that reference right? And not the annotations of the referenced type. May need to fix this to add the annotation to the type reference and not the tuple

Reply:
Yes. Can you add a sample test and check